### PR TITLE
AirfRANS: Fourier+3L/192d+T_max=12 — fine-grained T_max sweep

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

The T_max=10 cosine schedule creates a phase transition at epoch 40 (cosine trough of cycle 4), reaching val_primary/surface_mse = 0.0207. T_max=15 is being tested by winry (#2636). T_max=12 fills the narrow gap between these two values.

The phase transition timing is determined by when the LR reaches zero. For T_max=10 at ~750 steps/epoch, the 4th trough hits at step 4*10*750 = 30,000 steps ≈ epoch 40. For T_max=12, the 4th trough hits at step 4*12*750 = 36,000 steps ≈ epoch 48. If there is a deeper minimum accessible slightly later, T_max=12 may find it before T_max=15 would. This is a precision sweep of the most effective hyperparameter we have found.

## Instructions

Run the following command:

```bash
cd target/icml2026 && python train.py \
  --dataset airfrans \
  --airfrans_task full \
  --optimizer adamw \
  --lr 5e-4 \
  --cosine_t_max 12 \
  --no-use-ema \
  --enable-fourier \
  --epochs 999 \
  --wandb_group airfrans-tmax-sweep
```

Key change vs baseline:
- `--cosine_t_max 12` (up from 10, between baseline and winry's T_max=15)
- All other flags identical to the baseline T_max=10 run

**Critical:** The phase transition for T_max=12 should occur around epoch 48 (instead of epoch 40 for T_max=10). Run until at least epoch 55 to ensure you capture the transition. Report:
1. Best val_primary/surface_mse and at which epoch
2. Whether a sharp phase transition occurred and at what epoch
3. Comparison to T_max=10 transition depth (our 0.0207 baseline)

## Baseline

- **Dataset:** AirfRANS (full task)
- **Primary metric:** `val_primary/surface_mse` (lower is better)
- **Current best:** 0.0207 (val) — PR #2617 (T_max=10, lr=5e-4, 3L/192d, AdamW, epoch 40)
- **External target:** 0.0043 (~4.8x gap remaining)
- **Baseline W&B run:** z7t3ibwi
- **In-flight T_max experiments:** T_max=5 (thorfinn #2613), T_max=8 (roy #2639), T_max=15 (winry #2636)
- **Reproduce baseline:** `cd target/icml2026 && python train.py --dataset airfrans --airfrans_task full --optimizer adamw --lr 5e-4 --cosine_t_max 10 --no-use-ema --enable-fourier --epochs 999`

### CODE REVIEW

**Comment by tcapelle** [comment]
ADVISOR: Closing — T_max=12 without grad-clip is obsolete. Baseline now 0.01419 via grad-clip=1.0. itachi, please pick up your next assignment.
